src/commands/init.ts

```diff
@@ -0,0 +1,163 @@
+      description: 'Automatically accept default values for all prompts',
+    },
+    {
+      command: '<%= config.bin %> <%= command.id %> my-cli --module-type CommonJS --author "John Doe"',
+      description: 'Supply answers for specific prompts',
+    },
+    {
+      command: '<%= config.bin %> <%= command.id %> my-cli --module-type CommonJS --author "John Doe" --yes',
+      description: 'Supply answers for specific prompts and accept default values for the rest',
+    },
+  ]
+
+  static flaggablePrompts = FLAGGABLE_PROMPTS
+
+  static flags = {
+    ...makeFlags(FLAGGABLE_PROMPTS),
+    'module-type': Flags.option({
+      options: validModuleTypes,
+    })({}),
+  }
+
+  static summary = 'Initialize a new oclif CLI'
+
+  async run(): Promise<void> {
+    const outputDir = await this.getFlagOrPrompt({defaultValue: './', name: 'output-dir', type: 'input'})
+    const location = resolve(outputDir)
+
+    this.log(`Initializing oclif in ${chalk.green(location)}`)
+
+    const packageJSON = (await readPJSON(location))!
+    if (!packageJSON) {
+      throw new Errors.CLIError(`Could not find a package.json file in ${location}`)
+    }
+
+    const bin = await this.getFlagOrPrompt({
+      defaultValue: location.split(sep).at(-1)!,
+      name: 'bin',
+      type: 'input',
+    })
+
+    let moduleType = this.flags['module-type']
+    if (!moduleType || !validModuleTypes.includes(moduleType)) {
+      if (packageJSON.type === 'module') {
+        moduleType = 'ESM'
+      } else if (packageJSON.type === 'commonjs') {
+        moduleType = 'CommonJS'
+      } else {
+        moduleType = await (
+          await import('@inquirer/select')
+        ).default({
+          choices: validModuleTypes.map((type) => ({name: type, value: type})),
+          message: 'Select a module type:',
+        })
+      }
+    }
+
+    this.log(`Using module type ${chalk.green(moduleType)}`)
+
+    const template = moduleType === 'ESM' ? 'hello-world-esm' : 'hello-world'
+    const repoPath = join(location, '_tmp-template')
+    await clone(template, repoPath)
+
+    const repoBinPath = join(repoPath, 'bin')
+    const binFiles = await readdir(repoBinPath)
+    for (const binFile of binFiles) {
+      const filePath = join(repoBinPath, binFile)
+      if (statSync(filePath).isFile()) {
+        renameSync(filePath, join(location, 'bin', binFile))
+      }
+    }
+
+    await rm(repoPath, {recursive: true})
+
+    const updatedPackageJSON = {
+      ...packageJSON,
+      bin: {[bin]: './bin/run.js'},
+      oclif: {
+        ...packageJSON.oclif,
+        bin,
+        dirname: bin,
+      },
+    }
+
+    await writeFile(join(location, 'package.json'), JSON.stringify(updatedPackageJSON, null, 2))
+
+    const installedDeps = Object.keys(packageJSON.dependencies || {})
+    if (!installedDeps.includes('@oclif/core')) {
+      let packageManager = this.flags['package-manager']
+      if (!packageManager || !validPackageManagers.includes(packageManager)) {
```

Comment:
Same as module type, this allows us to accept the flag, then auto-detect, then ask if we don't have anything to go with.